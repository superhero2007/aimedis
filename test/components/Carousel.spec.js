import React from 'react';
import { mount } from 'enzyme';

import Carousel from 'components/Carousel';

function setup() {
  return mount(<Carousel />);
}

describe('Carousel', () => {
  const wrapper = setup();

  it('should be a Component', () => {
    expect(wrapper.instance() instanceof React.Component).toBe(true);
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();

    const prevButtonElement = wrapper.find('.app__carousel__indicator__prev');
    prevButtonElement.simulate('click');

    expect(wrapper.html()).toMatchSnapshot();

    const nextButtonElement = wrapper.find('.app__carousel__indicator__next');
    nextButtonElement.simulate('click');

    expect(wrapper.html()).toMatchSnapshot();
  });
});
