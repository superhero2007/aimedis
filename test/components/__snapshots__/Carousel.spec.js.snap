// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render properly 1`] = `"<div class=\\"app__carousel\\"><div class=\\"app__carousel__inner\\"><div class=\\"app__carousel__inner__item prev\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div class=\\"app__carousel__indicator__prev\\"><span class=\\"i-long-arrow-left\\"></span></div><div class=\\"app__carousel__inner__item__content\\"><img src=\\"\\" alt=\\"No Carousel\\"><p class=\\"app__carousel__inner__item__description\\">\\"Aimedis supports patients and doctors by making data available quickly and reliable which results in a quick, easy and efficient treatment.\\"</p><p class=\\"app__carousel__inner__item__name\\">Rizmy Shariff</p><p class=\\"app__carousel__inner__item__caption\\">Family doctor Solingen Germany</p><p class=\\"app__carousel__inner__item__quotation\\">“</p></div></div></div><div class=\\"app__carousel__inner__item active\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div class=\\"app__carousel__inner__item__content\\"><img src=\\"\\" alt=\\"No Carousel\\"><p class=\\"app__carousel__inner__item__description\\">\\"Aimedis supports patients and doctors by making data available quickly and reliable which results in a quick, easy and efficient treatment.\\"</p><p class=\\"app__carousel__inner__item__name\\">Rizmy Shariff</p><p class=\\"app__carousel__inner__item__caption\\">Family doctor Solingen Germany</p><p class=\\"app__carousel__inner__item__quotation\\">“</p></div></div></div><div class=\\"app__carousel__inner__item next\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div class=\\"app__carousel__inner__item__content\\"><img src=\\"\\" alt=\\"No Carousel\\"><p class=\\"app__carousel__inner__item__description\\">\\"Aimedis supports patients and doctors by making data available quickly and reliable which results in a quick, easy and efficient treatment.\\"</p><p class=\\"app__carousel__inner__item__name\\">Rizmy Shariff</p><p class=\\"app__carousel__inner__item__caption\\">Family doctor Solingen Germany</p><p class=\\"app__carousel__inner__item__quotation\\">“</p></div><div class=\\"app__carousel__indicator__next\\"><span class=\\"i-long-arrow-right\\"></span></div></div></div></div></div>"`;

exports[`Carousel should render properly 2`] = `"<div class=\\"app__carousel\\"><div class=\\"app__carousel__inner\\"><div class=\\"app__carousel__inner__item prev\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div class=\\"app__carousel__indicator__prev\\"><span class=\\"i-long-arrow-left\\"></span></div><div class=\\"app__carousel__inner__item__content\\"><img src=\\"\\" alt=\\"No Carousel\\"><p class=\\"app__carousel__inner__item__description\\">\\"Aimedis supports patients and doctors by making data available quickly and reliable which results in a quick, easy and efficient treatment.\\"</p><p class=\\"app__carousel__inner__item__name\\">Rizmy Shariff</p><p class=\\"app__carousel__inner__item__caption\\">Family doctor Solingen Germany</p><p class=\\"app__carousel__inner__item__quotation\\">“</p></div></div></div><div class=\\"app__carousel__inner__item active\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div class=\\"app__carousel__inner__item__content\\"><img src=\\"\\" alt=\\"No Carousel\\"><p class=\\"app__carousel__inner__item__description\\">\\"Aimedis supports patients and doctors by making data available quickly and reliable which results in a quick, easy and efficient treatment.\\"</p><p class=\\"app__carousel__inner__item__name\\">Rizmy Shariff</p><p class=\\"app__carousel__inner__item__caption\\">Family doctor Solingen Germany</p><p class=\\"app__carousel__inner__item__quotation\\">“</p></div></div></div><div class=\\"app__carousel__inner__item next\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div class=\\"app__carousel__inner__item__content\\"><img src=\\"\\" alt=\\"No Carousel\\"><p class=\\"app__carousel__inner__item__description\\">\\"Aimedis supports patients and doctors by making data available quickly and reliable which results in a quick, easy and efficient treatment.\\"</p><p class=\\"app__carousel__inner__item__name\\">Rizmy Shariff</p><p class=\\"app__carousel__inner__item__caption\\">Family doctor Solingen Germany</p><p class=\\"app__carousel__inner__item__quotation\\">“</p></div><div class=\\"app__carousel__indicator__next\\"><span class=\\"i-long-arrow-right\\"></span></div></div></div></div></div>"`;

exports[`Carousel should render properly 3`] = `"<div class=\\"app__carousel\\"><div class=\\"app__carousel__inner\\"><div class=\\"app__carousel__inner__item prev\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div class=\\"app__carousel__indicator__prev\\"><span class=\\"i-long-arrow-left\\"></span></div><div class=\\"app__carousel__inner__item__content\\"><img src=\\"\\" alt=\\"No Carousel\\"><p class=\\"app__carousel__inner__item__description\\">\\"Aimedis supports patients and doctors by making data available quickly and reliable which results in a quick, easy and efficient treatment.\\"</p><p class=\\"app__carousel__inner__item__name\\">Rizmy Shariff</p><p class=\\"app__carousel__inner__item__caption\\">Family doctor Solingen Germany</p><p class=\\"app__carousel__inner__item__quotation\\">“</p></div></div></div><div class=\\"app__carousel__inner__item active\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div class=\\"app__carousel__inner__item__content\\"><img src=\\"\\" alt=\\"No Carousel\\"><p class=\\"app__carousel__inner__item__description\\">\\"Aimedis supports patients and doctors by making data available quickly and reliable which results in a quick, easy and efficient treatment.\\"</p><p class=\\"app__carousel__inner__item__name\\">Rizmy Shariff</p><p class=\\"app__carousel__inner__item__caption\\">Family doctor Solingen Germany</p><p class=\\"app__carousel__inner__item__quotation\\">“</p></div></div></div><div class=\\"app__carousel__inner__item next\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div class=\\"app__carousel__inner__item__content\\"><img src=\\"\\" alt=\\"No Carousel\\"><p class=\\"app__carousel__inner__item__description\\">\\"Aimedis supports patients and doctors by making data available quickly and reliable which results in a quick, easy and efficient treatment.\\"</p><p class=\\"app__carousel__inner__item__name\\">Rizmy Shariff</p><p class=\\"app__carousel__inner__item__caption\\">Family doctor Solingen Germany</p><p class=\\"app__carousel__inner__item__quotation\\">“</p></div><div class=\\"app__carousel__indicator__next\\"><span class=\\"i-long-arrow-right\\"></span></div></div></div></div></div>"`;
